packages/merge-on-green/src/merge-on-green.ts

```diff
@@ -331,6 +331,98 @@ handler.checkPRMergeability = async function checkPRMergeability(
   }
 };
 
+/**
+ * For a given repository, looks through all the PRs and checks to see if they have a MOG label
+ * @param owner the owner of the repo
+ * @param repo the repo name
+ * @param context the context of the webhook payload
+ * @returns void
+ */
+handler.pickUpPR = async function pickUpPR(
+  owner: string,
+  repo: string,
+  github: Context['github']
+) {
+  const prs = await github.paginate(await github.pulls.list, {
+    owner,
+    repo,
+    state: 'open',
+  });
+
+  for (const pr of prs) {
+    const labels = await github.paginate(
+      await github.issues.listLabelsOnIssue,
+      {
+        owner,
+        repo,
+        issue_number: pr.number,
+      }
+    );
+
+    const labelFound = labels.find(
+      (label: Label) =>
+        label.name === MERGE_ON_GREEN_LABEL ||
+        label.name === MERGE_ON_GREEN_LABEL_SECURE
+    );
+
+    if (labelFound) {
```

Comment:
we should only do this if there's not already an entry in `datastore` for the given PR right?